pulp_smash/tests/rpm/api_v2/test_unassociate.py

```diff
@@ -92,50 +100,28 @@ def setUpClass(cls):
         sync_path = urljoin(repo['_href'], 'actions/sync/')
         client.post(sync_path, {'override_config': {}})
 
-        # List starting content
-        cls.before_units = {
-            type_id: _list_repo_units_of_type(client, repo['_href'], type_id)
-            for type_id in cls.TYPE_IDS
-        }
-
-        # Remove one of each unit and store its id for later assertions
-        cls.removed_units = {}
-        for type_id, units_list in cls.before_units.items():
-            cls.removed_units[type_id] = units_list[0]
-            _remove_unit(client, repo['_href'], type_id, units_list[0])
-
-        # List final content
-        cls.after_units = {
-            type_id: _list_repo_units_of_type(client, repo['_href'], type_id)
-            for type_id in cls.TYPE_IDS
-        }
-
-    def test_units_before_removal(self):
-        """Assert the removed units are present before the removal."""
-        for type_id in self.TYPE_IDS:
-            with self.subTest(type_id=type_id):
-                self.assertIn(
-                    self.removed_units[type_id],
-                    self.before_units[type_id],
-                )
-
-    def test_units_after_removal(self):
-        """Assert the removed units are not present after the removal."""
-        for type_id in self.TYPE_IDS:
-            with self.subTest(type_id=type_id):
-                self.assertNotIn(
-                    self.removed_units[type_id],
-                    self.after_units[type_id],
-                )
-
-    def test_one_unit_removed(self):
+        # Remove one unit of each type.
+        cls.units_before = _search_units(cls.cfg, repo['_href'], cls.TYPE_IDS)
+        cls.units_removed = []
+        for type_id in cls.TYPE_IDS:
+            unit = random.choice(_get_units_by_type(cls.units_before, type_id))
+            cls.units_removed.append(unit)
+            _remove_unit(cls.cfg, repo['_href'], unit)
+        cls.units_after = _search_units(cls.cfg, repo['_href'], cls.TYPE_IDS)
+
+    def test_units_not_in_search(self):
+        """Assert the removed units do not appear in search results."""
+        ids_removed = {_get_unit_id(unit) for unit in self.units_removed}
+        ids_after = {_get_unit_id(unit) for unit in self.units_after}
+        self.assertEqual(ids_removed & ids_after, set())  # s1.isdisjoint(s2)
```

Comment:
Smart move, using assertTrue and isdisjoint will give the same result, but using assertEqual with the two sets will actually prints what are the differences. :100: